File: Proga/Lab6/Lab6/NumString.cpp
#include "NumString.h"
#include <ctype.h>

// Public //

NumString::NumString(const char* strIn)
{
    size_t length = strNumLen(strIn);
    this->str = new char[length + 1];
    size_t j = 0;
    for (size_t i = 0; i < strlen(strIn); i++)
    {
        if (isdigit(strIn[i]))
            this->str[j++] = strIn[i];
    }
    this->str[length] = '\0';
}

NumString::NumString(char* strIn)
{
    size_t length = strNumLen(strIn);
    this->str = new char[length + 1];
    size_t j = 0;
    for (size_t i = 0; i < strlen(strIn); i++)
    {
        if (isdigit(strIn[i]))
            this->str[j++] = strIn[i];
    }
    this->str[length] = '\0';
}

void NumString::MoveLetters()
{
    size_t length = strlen(this->str);
    size_t amount = 1;
    char* temp = new char[amount];
    for (size_t i = 0; i < amount; i++)
    {
        temp[i] = this->str[i];
    }
    for (size_t i = 0; i < length - amount; i++)
    {
        this->str[i] = this->str[i + amount];
    }
    for (size_t i = 0; i < amount; i++)
    {
        this->str[length - 1 - i] = temp[amount - 1 - i];
    }
    delete[] temp;
}

size_t NumString::GetLength()
{
    return strNumLen(this->str);
}

// Private //

size_t NumString::strNumLen(char* strIn)
{
    size_t len = strlen(strIn);
    size_t counter = 0;
    for (size_t i = 0; i < len; i++)
    {
        if (isdigit(strIn[i]))
            counter++;
    }
    return counter;
}

size_t NumString::strNumLen(const char* strIn)
{
    size_t len = strlen(strIn);
    size_t counter = 0;
    for (size_t i = 0; i < len; i++)
    {
        if (isdigit(strIn[i]))
            counter++;
    }
    return counter;
}

File: Proga/Lab6/MyClasses/NumString.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MyClasses
{
    public class NumString : MyString
    {
        public NumString(in string strIn)
        {
            int length = strNumLen(strIn);
            this.str = new char[length];
            int j = 0;
            for (int i = 0; i < strIn.Length; i++)
            {
                if (Char.IsDigit(strIn[i]))
                    this.str[j++] = strIn[i];
            }  
        }
        public override void MoveLetters()
        {
            int length = this.str.Length;
            int amount = 1;
            char[] temp = new char[amount];
            for (int i = 0; i < amount; i++)
            {
                temp[i] = this.str[i];
            }
            for (int i = 0; i < length - amount; i++)
            {
                this.str[i] = this.str[i + amount];
            }
            for (int i = 0; i < amount; i++)
            {
                this.str[length - 1 - i] = temp[amount - 1 - i];
            }
        }
        public override int GetLength()
        {
            return strNumLen(this.str);
        }
        private int strNumLen(in char[] strIn)
        {
            int len = strIn.Length;
            int counter = 0;
            for (int i = 0; i < len; i++)
            {
                if (Char.IsDigit(strIn[i]))
                    counter++;
            }
            return counter;
        }
        private int strNumLen(in string strIn)
        {
            int len = strIn.Length;
            int counter = 0;
            for (int i = 0; i < len; i++)
            {
                if (Char.IsDigit(strIn[i]))
                    counter++;
            }
            return counter;
        }
    }
}
